server/parser/src/main/antlr4/org/eclipse/lsp/cobol/core/CobolParser.g4

```diff
@@ -2181,18 +2173,20 @@ length
    ;
 
 argument
-   : arithmeticExpression
-   | TRAILING | LEADING
+   : ALL
+   | arithmeticExpression
+   | TRAILING
+   | LEADING
    ;
 
 // qualified data name ----------------------------------
 
 qualifiedDataName
-   : variableUsageName tableCall? referenceModifier? inData*
+   : FUNCTION? variableUsageName tableCall? referenceModifier? inData*
```

Comment:
`qualifiedDataName` is used all over the place. Is the function prefix valid in all those contexts?

Reply:
You are right. I know for sure that variable used in the procedure division (like PROCEDURE DIVISION USING ...)
should not allow function call